### ISSUE TITLE
The first example in "Getting started" results in 404

### ISSUE THREAD

**Issue by srikumarks** [AnswerDotAI/fasthtml#174]
So I'm running python 3.12.4 and wanting to checkout FastHtml and I put the following in `main.py` -

```py
from fasthtml.common import *

app,rt = fast_app()

@rt('/')
def get():
    return Div(P('hello world!'))

serve()
```

and do `python main.py` and I get a `404 not found` when I visit `http://localhost:5001`. The server (uvicorn) shows no errors on startup and the output of the running `python main.py` is below -

```
Link: http://localhost:5001
INFO:     Will watch for changes in these directories: ['/************hiding-path*****']
INFO:     Uvicorn running on http://0.0.0.0:5001 (Press CTRL+C to quit)
INFO:     Started reloader process [3845] using WatchFiles
INFO:     Started server process [3862]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
```

I tried variants like mounting on `@rt('/hello')` and all resulted in 404. When I did this step by step from the command line and checked `help(rt)`, it showed a message that seemed to indicate a deprecation of the decorator usage within Starlette -

```
Help on method route in module fasthtml.core:

route(path: str, methods=None, name=None, include_in_schema=True) method of fasthtml.core.FastHTML instance
    We no longer document this decorator style API, and its usage is discouraged.
    Instead you should use the following approach:

    >>> routes = [Route(path, endpoint=...), ...]
    >>> app = Starlette(routes=routes)
```

Not sure if that has anything to do with it. Haven't investigated further since I just wanted to checkout fasthtml. A similar basic instruction with FastAPI which serves a JSON worked fine, so I suppose that implies no issues with network or binding or any such thing.

**Comment by pydanny** [comment]
Hey @srikumarks, I can't replicate your problem, even on Python 3.12. Can you tell share your operating system?

**Comment by srikumarks** [comment]
OS is MacOS Sonoma 14.5 running on an M1 MacBook Air.

Btw I'm seeing another intro to FastHtml that gives the following code - 

```py
from fasthtml.common import *

app = FastHTML()

@app.get("/")
def hello():
    return "<h1>Hello world</h1>"
```

.. which (when saved to `app.py`) is expected to be launched as `uvicorn app:app --reload` ... and this works on my machine/OS. Is this like a newer and preferred API?

**Comment by srikumarks** [comment]
The original example I tried was from Jeremy Howard's writeup - https://www.answer.ai/posts/2024-08-03-fasthtml.html and this other one is from the docs - https://docs.fastht.ml/tutorials/by_example.html

**Comment by pydanny** [comment]
Odd! Let's figure this out! Can you try this:

```python
from fasthtml.fastapp import *

app, rt = fast_app()

@rt("/")
def get():
    return Titled("FastHTML", P("Let's do this!"))

serve()
```

> .. which (when saved to `app.py`) is expected to be launched as `uvicorn app:app --reload` ... and this works on my machine/OS. Is this like a newer and preferred API?

`fast_app` is the newer API, but it wraps the `FastHTML` loader and returns a tuple of `app` and `rt`. There's nothing magical about it. 🤔 

> ... and doing `return Div(P("hello world"))` also works, which makes this the canonical version for me I guess?

Have you tried `127.0.0.1` or `0.0.0.0`? I've seen some people have issues with those, but I couldn't tell you why they do.

**Comment by srikumarks** [comment]
Just tried. Same `404 not found` result with this too.

**Comment by srikumarks** [comment]
Ok one finding that could be useful. If I remove the last `serve()` line and run the program with `uvicorn main:app --reload`, it works. But uvicorn doesn't bind to 0.0.0.0, but to 127.0.0.1, so I changed the `serve()` line to read -

```py
serve(host='127.0.0.1', port=5002)
```

... and et voila! It worked! So perhaps some network setting it is that's preventing binding to 0.0.0.0. It's perhaps a good idea to bind to 127.0.0.1 by default btw, just like how uvicorn does.

**Comment by srikumarks** [comment]
An odd thing I noted was that though I had no python programs running, when I'd run on the default 5001 port and then made the `host=127.0.0.1` setting, I kept getting a complaint that the port is under use despite no server programs running. So I had to change the port to 5002 to get it to work.

**Comment by jph00** [comment]
Glad you got it fixed! The reason we default to 0.0.0.0 is that some deployment providers require that.
